src/i965_avc_encoder_common.c

```diff
@@ -153,108 +135,40 @@ i965_avc_calculate_initial_qp(struct avc_param * param)
 int
 i965_avc_get_max_v_mv_r(int level_idc)
 {
-    int max_v_mv_r = 128 * 4;
-
-    // See JVT Spec Annex A Table A-1 Level limits for below mapping
-    // MaxVmvR is in luma quarter pel unit
-    switch (level_idc) {
-    case INTEL_AVC_LEVEL_1:
-        max_v_mv_r = 64 * 4;
-        break;
-    case INTEL_AVC_LEVEL_11:
-    case INTEL_AVC_LEVEL_12:
-    case INTEL_AVC_LEVEL_13:
-    case INTEL_AVC_LEVEL_2:
-        max_v_mv_r = 128 * 4;
-        break;
-    case INTEL_AVC_LEVEL_21:
-    case INTEL_AVC_LEVEL_22:
-    case INTEL_AVC_LEVEL_3:
-        max_v_mv_r = 256 * 4;
-        break;
-    case INTEL_AVC_LEVEL_31:
-    case INTEL_AVC_LEVEL_32:
-    case INTEL_AVC_LEVEL_4:
-    case INTEL_AVC_LEVEL_41:
-    case INTEL_AVC_LEVEL_42:
-    case INTEL_AVC_LEVEL_5:
-    case INTEL_AVC_LEVEL_51:
-    case INTEL_AVC_LEVEL_52:
-        max_v_mv_r = 512 * 4;
-        break;
-    default:
-        assert(0);
```

Comment:
How about print a warning message since the assertion has been removed?

Reply:
I think that warning would be better placed further out - probably somewhere in the H.264 encoder setup, maybe gen9_avc_update_parameters()?  I'll refactor to add an "h264_level_is_valid" function and use it there.

Reply:
Yes, would better don't place the warning here. But I think put it in  intel_encoder_check_avc_parameter should be better.  BTW, do we need to reset it to be a default level such as 2 or just let the invalid level_idc written to the bitream? level 2 is what is using for  i965_avc_get_max_mbps and i965_avc_get_max_v_mv_r.
As a summary of my point, we can have two choices:
1. Print a warning message and reset it to be level 2. 
2. Just return an error code of  VA_STATUS_ERROR_INVALID_PARAMETER in intel_encoder_check_avc_parameter.

Reply:
I don't think we want to do either of those.  The level_idc value is already written to the bitstream in the packed header, so it can't be changed.  It's also not necessarily invalid, even if it's not recognised - it could be a new version of the standard or an extension (consider the current problem, where a newer version of the standard defines new levels and the old driver aborts when it sees them).  I think just a simple warning and guessing some sensible limit values is the best approach.

Reply:
Got it. At least one thing needs to be changed, gen9_avc_update_parameters is not the best place to     give the warning. It means similar thing need to be done for gen6/7/8/x.

Reply:
Currently only gen9 performs this test - gen<9 is not affected at all by the change.  If that is wanted, it could be done separately.

Reply:
now gen8 and gen9 share the same code for AVC encoding, so this test is for both gen8 and gen9. BTW i965_avc_get_max_v_mv_r() is not called at all, how about to remove this function at all?